### PR TITLE
Ability to make missed_count optional

### PR DESCRIPTION
**Pull request by KartulUdus**
## Description

This PR gives the ability to set a custom missed_count for a spawnpoint to be disabled.
Alternatively disable it entirely with `-mc 0` | `--missed-count 0`

## Motivation and Context

When changing accounts after blinding, some spawnpoints have been missed over 5 times. 
Some regularly run sql to set missed_count to 0.
Doing less repedetive tasks helps with quality of life. with this pr, spawnpoint disabling can be disabled. 

## How Has This Been Tested?

1 worker in Helsinki now, presumably results after blinded. 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/first-run/commandline.md [hunk 1]
```diff
@@ -396,13 +396,17 @@ environment variables which override config file values which override defaults.
...
+      -mc MISSED_COUNT, --missed-count MISSED_COUNT
+                            Amount of times spawnpoint is missed before before
+                            it's disabled (Default: 5). 0 to disable. [env var:
+                            POGOMAP_MISSED_COUNT]
       -v                    Show debug messages from RocketMap and pgoapi. Can be
                             repeated up to 3 times.
       --verbosity VERBOSE   Show debug messages from RocketMap and pgoapi. [env
                             var: POGOMAP_VERBOSITY]
-
+
```
Comment (pogo-excalibur) [inline docs/first-run/commandline.md]: This probably should have caused a Travis fail -> ❌

File: docs/first-run/commandline.md [hunk 2]
```diff
@@ -414,7 +418,7 @@ environment variables which override config file values which override defaults.
       --db-threads DB_THREADS
                             Number of db threads; increase if the db queue falls
                             behind. [env var: POGOMAP_DB_THREADS]
-
+
```
Comment (pogo-excalibur) [inline docs/first-run/commandline.md]: Same as the above for this -> ❌

File: pogom/models.py [hunk 3]
```diff
@@ -1332,7 +1332,7 @@ def get_times(cell, scan, now_date, scan_delay,
 
         for sp in linked_spawn_points:
 
-            if sp['missed_count'] > 5:
+            if sp['missed_count'] > args.missed_count and args.missed_count:
```
Comment (pogo-excalibur) [inline pogom/models.py]: 🤔 Hm.

File: docs/first-run/commandline.md [hunk 4]
```diff
@@ -396,13 +396,17 @@ environment variables which override config file values which override defaults.
       --dump                Dump censored debug info about the environment and
                             auto-upload to hastebin.com. [env var: POGOMAP_DUMP]
       -exg, --ex-gyms       Fetch OSM parks within geofence and flag gyms that are
-                            candidates for ex raids. Only required once per area.
+                            candidates for EX raids. Only required once per area.
                             [env var: POGOMAP_EX_GYMS]
+      -mc MISSED_COUNT, --missed-count MISSED_COUNT
```
Comment (pogo-excalibur) [inline docs/first-run/commandline.md]: You'll want to set the default value here.

File: pogom/models.py (hunk 3 above)
Comment (pogo-excalibur) [reply pogom/models.py]: If `args.missed_count` defaults to 5, then the ` and args.missed_count` is unnecessary.

File: pogom/models.py [hunk 5]
```diff
@@ -2277,7 +2277,7 @@ def parse_map(args, map_dict, scan_coords, scan_location, db_update_queue,
     # Look for spawnpoints within scan_loc that are not here to see if we
     # can narrow down tth window.
     for sp in ScannedLocation.linked_spawn_points(scan_location['cellid']):
-        if sp['missed_count'] > 5:
+        if sp['missed_count'] > args.missed_count and args.missed_count:
```
Comment (pogo-excalibur) [inline pogom/models.py]: Repeat of the above comment.

File: pogom/schedulers.py [hunk 6]
```diff
@@ -768,7 +768,8 @@ def schedule(self):
                 spawnpoints = SpawnPoint.select_in_hex_by_cellids(
                     self.scans.keys(), self.location_change_date)
                 for sp in spawnpoints:
-                    if sp['missed_count'] > 5:
+                    if (sp['missed_count'] > self.args.missed_count and
```
Comment (pogo-excalibur) [inline pogom/schedulers.py]: Same as the above comment.

File: pogom/models.py (hunk 3 above)
Comment (KartulUdus) [reply pogom/models.py]: It's there so it could be set to 0 (== false)

File: pogom/utils.py [hunk 7]
```diff
@@ -501,6 +501,11 @@ def get_args():
                               'gyms that are candidates for EX raids. ' +
                               'Only required once per area.'),
                         action='store_true', default=False)
+    parser.add_argument('-mc', '--missed-count',
+                        help=('Amount of times spawnpoint is missed before ' +
```
Comment (pogo-excalibur) [inline pogom/utils.py]: Perhaps:
```
Number of times that a spawnpoint can be missed before it gets disabled.
```

File: docs/first-run/commandline.md (hunk 2 above)
Comment (KartulUdus) [reply docs/first-run/commandline.md]: 😬

File: pogom/models.py (hunk 3 above)
Comment (pogo-excalibur) [reply pogom/models.py]: Right. And putting that part of the condition first would allow proper short-circuiting if 'missed_count' is disabled.

Comment (KartulUdus) [reply pogom/models.py]: so
`if args.missed_count and sp['missed_count'] > args.missed_count:`
?

Comment (pogo-excalibur) [reply pogom/models.py]: I'd prefer the brackets for clarity, but that'd be the right order.

File: pogom/utils.py [hunk 8]
```diff
@@ -501,6 +501,11 @@ def get_args():
                               'gyms that are candidates for EX raids. ' +
                               'Only required once per area.'),
                         action='store_true', default=False)
+    parser.add_argument('-mc', '--missed-count',
+                        help=('Number of times that a spawnpoint can be' +
```
Comment (pogo-excalibur) [inline pogom/utils.py]: Add a space at the end of the line.

Comment (KartulUdus) [reply pogom/utils.py]: ✅

**Comment by KartulUdus** [comment]
think it's pretty safe to close this pr

**Pull request closed without merging**